sparse/coo/core.py

```diff
@@ -1572,28 +1560,294 @@ def _keepdims(original, new, axis):
     return new.reshape(shape)
 
 
-def _mask(coords, idx, shape):
-    if isinstance(idx, numbers.Integral):
-        return coords == idx
-    elif isinstance(idx, slice):
-        step = idx.step if idx.step is not None else 1
-        if step > 0:
-            start = idx.start if idx.start is not None else 0
-            stop = idx.stop if idx.stop is not None else shape
-            return (coords >= start) & (coords < stop) & \
-                   (coords % step == start % step)
+def _mask(coords, indices, shape):
```

Comment:
Request small docstring here.  Reviewing this I don't know what you mean by a mask.  I can take a guess, but it's unlikely that someone who has to fix bugs in this code in the future will understand immediately.

Reply:
Maybe it would make sense to move all of these functions to a separate file (slicing.py?) and add a module-level doscstring describing how they work together?

Reply:
This sounds like a good idea, but `slicing.py` isn't suitable as it's common to `COO` and `DOK`.

Reply:
A different name then?